temporalio/worker/_workflow.py

```diff
@@ -548,6 +550,7 @@ def _create_workflow_instance(
             extern_functions=self._extern_functions,
             disable_eager_activity_execution=self._disable_eager_activity_execution,
             worker_level_failure_exception_types=self._workflow_failure_exception_types,
+            assert_activity_valid=self._assert_activity_valid,
```

Comment:
Going through the sandbox with a callable directly is a bit rough. Technically it works because it's a simple string and exception, but I would prefer if we used `_extern_functions`. See how we did `__temporal_get_metric_meter` there. It just provides a good separation of known extern functions into the sandbox instead of adding to the end of details (there is no analog in .NET which doesn't have a sandbox).